ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml

```diff
@@ -6,63 +6,123 @@ import StatusQ.Controls 0.1
 
+    required property var baseTokensModel
+    required property var baseCollectiblesModel
+
+    readonly property bool dirty: {
+        if (!loader.item)
+            return false
+        if (tabBar.currentIndex > 1)
+            return false
+        if (tabBar.currentIndex === 1 && (baseCollectiblesModel.isUpdating || baseCollectiblesModel.isFetching))
+            return false
+        return loader.item && loader.item.dirty
+    }
+
+    function saveChanges() {
+        if (tabBar.currentIndex > 1)
+            return
+        loader.item.saveSettings()
+    }
+
+    function resetChanges() {
+        if (tabBar.currentIndex > 1)
+            return
+        loader.item.revert()
+    }
+
+    QtObject {
+        id: d
+
+        function checkLoadMoreCollectibles() {
+            if (tabBar.currentIndex !== 1)
+                return
+            // If there is no more items to load or we're already fetching, return
+            if (!root.baseCollectiblesModel.hasMore || root.baseCollectiblesModel.isFetching)
+                return
+            root.baseCollectiblesModel.loadMore()
+        }
+    }
+
+    Connections {
+        target: root.baseCollectiblesModel
+        function onHasMoreChanged() {
+            d.checkLoadMoreCollectibles()
+        }
+        function onIsFetchingChanged() {
+            d.checkLoadMoreCollectibles()
+        }
+    }
+
     StatusTabBar {
         id: tabBar
 
         Layout.fillWidth: true
         Layout.topMargin: 5
 
         StatusTabButton {
-            id: assetsTab
+            leftPadding: 0
             width: implicitWidth
             text: qsTr("Assets")
         }
 
         StatusTabButton {
-            id: collectiblesTab
             width: implicitWidth
             text: qsTr("Collectibles ")
         }
 
         StatusTabButton {
-            id: tokensListTab
             width: implicitWidth
             text: qsTr("Token lists")
         }
     }
 
-    StackLayout {
-        id: stackLayout
-
+    // NB: we want to discard any pending unsaved changes when switching tabs or navigating away
+    Loader {
+        id: loader
         Layout.fillWidth: true
         Layout.fillHeight: true
-        currentIndex: tabBar.currentIndex
+        active: visible
+
+        sourceComponent: {
+            switch (tabBar.currentIndex) {
+            case 0:
+                return tokensPanel
+            case 1:
+                return collectiblesPanel
+            case 2:
+                return supportedTokensListPanel
+            }
+        }
+    }
 
-        ShapeRectangle {
-            Layout.alignment: Qt.AlignHCenter
-            Layout.preferredWidth: parent.width - 4 // The rectangular path is rendered outside
-            Layout.maximumHeight: 44
-            text: qsTr("You’ll be able to manage the display of your assets here")
+    Component {
+        id: tokensPanel
+        ManageTokensPanel {
```

Comment:
Seeing them all together I realize this should be like this:
```suggestion
        ManageAssetsPanel {
```

Reply:
Yup... probably yes; I can rename them; the main wallet view also has "AssetsView/CollectiblesView"

Reply:
Done, renamed all! :)

Reply:
Thanks!